slither/analyses/data_dependency/data_dependency.py

```diff
@@ -26,39 +30,51 @@
     TemporaryVariableSSA,
     TupleVariableSSA,
 )
-from slither.core.solidity_types.type import Type
+from slither.slithir.variables.variable import SlithIRVariable
 
 if TYPE_CHECKING:
     from slither.core.compilation_unit import SlitherCompilationUnit
 
-
 ###################################################################################
 ###################################################################################
 # region User APIs
 ###################################################################################
 ###################################################################################
 
 
-Variable_types = Union[Variable, SolidityVariable]
+SUPPORTED_TYPES = Union[Variable, SolidityVariable]
+
+# TODO refactor the data deps to be better suited for top level function object
```

Comment:
Can we open an issue outlining what is currently wrong and the desired behavior?

Reply:
Yes, see the PR message: `This is a quick fix, and requires a larger refactoring of the data deps. A separate issue must be created`